bpfassets/libbpf/src/kepler.bpf.c

```diff
@@ -223,18 +235,7 @@ int kepler_sched_switch_trace(u64 *ctx)
 	// counter events to keep the metrics map current.
 	collect_metrics_and_reset_counters(&buf, prev_pid, curr_ts, cpu_id);
 
-	// Skip some samples to minimize overhead
-	// Note that we can only skip samples after updating the metric maps to
-	// collect the right values
-	if (SAMPLE_RATE > 0) {
-		if (counter_sched_switch > 0) {
-			counter_sched_switch--;
-			return 0;
-		}
-		counter_sched_switch = SAMPLE_RATE;
-	}
-
-	if (prev_state == TASK_RUNNING) {
+	if (prev_state | TASK_RUNNING) {
```

Comment:
We must only collect metrics if the task was running on the CPU. We want to gather CPU metrics (hardware counters and CPU time) exclusively when the task was actively running on the CPU.

This is important because if task1 goes from idle to running (in `sched_switch`) and we collect metrics from CPU1 while task3 was actually running on the CPU, we will mistakenly attribute the metrics of task3 to task1. To ensure accuracy, metrics collection should occur only for tasks that were running on the CPU.